client/mapView/MapView.cpp

```diff
@@ -115,9 +123,44 @@ void MapView::onMapScrolled(const Point & distance)
 
 void MapView::onMapSwiped(const Point & viewPosition)
 {
+	swipeHistory[GH.input().getTicks()] = viewPosition;
+
 	controller->setViewCenter(model->getMapViewCenter() + viewPosition, model->getLevel());
 }
 
+void MapView::postSwipe(uint32_t msPassed) {
+	if(!actions->dragActive)
+	{
+		if(swipeHistory.size() > 1)
+		{
+			Point diff = Point(0, 0);
+			std::pair<uint32_t, Point> firstAccepted;
+			uint32_t now = GH.input().getTicks();
+			for (auto & x : swipeHistory) {
+				if(now - x.first < 150) { // only the last 150 ms are catched
+					if(firstAccepted.first == 0)
+						firstAccepted = x;
+					diff += x.second;
+				}
+			}
+
+			uint32_t timediff = swipeHistory.rbegin()->first - firstAccepted.first;
+
+			postSwipeAngle = diff.angle();
+			postSwipeSpeed = static_cast<double>(diff.length()) / static_cast<double>(timediff); // unit: pixel/millisecond			
+		}
+		swipeHistory.clear();
+	}
+	if(postSwipeSpeed > 0.1) {
+		double len = postSwipeSpeed * static_cast<double>(msPassed);

```

Comment:
what if FPS is quite low so msPassed is large? For example, what if you have breakpoint in client while moving map?

Reply:
Should work as indented (`pixel/millisecond` -> `pixel`). The reason for this is exactly fps independence.

Reply:
Specifically to avoid issues with debugger, tick() and as result - this method will never receive value greater than 100 ms